### PR TITLE
feature/112400475-kc

### PR DESCRIPTION
**Pull request by kkchu791**
https://www.pivotaltracker.com/projects/1516637/stories/112400475

Add username field

When inviting a friend, if the friend is not already a user, a random username will be generated
for him from the Word model

### CODE REVIEW

File: app/models/word.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class Word < ActiveRecord::Base
-end
```
Comment (charliemcelfresh) [inline app/models/word.rb]: @kkchu791 let's not load all the words into memory. You can find how many words are in this table, then randomly grab two from it, let's try to do something like that.

File: app/views/users/_new_form.html.erb [hunk 2]
```diff
@@ -10,6 +10,11 @@
     </div>
   <% end %>
   <p>
```
Comment (charliemcelfresh) [inline app/views/users/_new_form.html.erb]: There are two ways that a user can get created. First, where the user comes and creates an account with the new user form. In this case, we don't need this messaging, and the user should be forced to create a username. So there should be a validation that makes this happen, on the User model.

Second is the case where the user does not exist, and someone invites the user to write a Haiku. This is the case where we create a username for the user, using your random_words method.

File: app/models/word.rb [hunk 3]
```diff
@@ -1,2 +1,9 @@
...
\ No newline at end of file
+  def random_words
```
Comment (charliemcelfresh) [inline app/models/word.rb]: This is fine the way you have it, but let's make it a class method.

**Comment by charliemcelfresh** [comment]
@kkchu791 this looks good! Let's also add two tests: 1) a request spec that covers the situation where the username field is blank, and 2) a test in the haikus_spec, that makes sure the username is created when we create a user account for a friend who a) is not yet a user, and b) is invited to create a haiku

File: spec/requests/haikus_spec.rb [hunk 4]
```diff
@@ -114,5 +114,22 @@
         expect(response).to redirect_to(log_in_url)
       end
     end
+
+    context 'when inviting a friend to write a haiku' do
```
Comment (thomas-jung) [inline spec/requests/haikus_spec.rb]: I think this should be moved to users_spec.rb since it doesn't have anything to do with haikus controller

**Comment by thomas-jung** [comment]
Good job Kirk! I like how you set up your factory too. Everything works. :+1: after you move that test into the users spec.

**Pull request merged**